doc/develop/toolchains/zephyr_sdk.rst

```diff
@@ -20,32 +23,39 @@ The Zephyr SDK supports the following target architectures:
 * x86 (32-bit and 64-bit)
 * Xtensa
 
+.. _toolchain_zephyr_sdk_bundle_variables:
+
+Installation bundle and variables
+*********************************
+
 The Zephyr SDK bundle supports all major operating systems (Linux, macOS and
 Windows) and is delivered as a compressed file.
 The installation consists of extracting the file and running the included setup
 script. Additional OS-specific instructions are described in the sections below.
 
-By default, the Zephyr build system assumes that the toolchain will be obtained
-from the Zephyr SDK. You can enforce this by setting the environment variable
-:envvar:`ZEPHYR_TOOLCHAIN_VARIANT` to ``zephyr``.
-
-If you install the Zephyr SDK outside any of the default locations (listed in
-the operating system specific instructions below), you must register the Zephyr
-SDK in the CMake package registry by running the setup script or setting
-:envvar:`ZEPHYR_SDK_INSTALL_DIR` to point to the Zephyr SDK installation
-directory.
-
-You can also set :envvar:`ZEPHYR_SDK_INSTALL_DIR` to point to a directory
-containing multiple Zephyr SDKs, allowing for automatic toolchain selection. For
-example, you can set ``ZEPHYR_SDK_INSTALL_DIR`` to ``/company/tools``, where the
-``company/tools`` folder contains the following subfolders:
-
-* ``/company/tools/zephyr-sdk-0.13.2``
-* ``/company/tools/zephyr-sdk-a.b.c``
-* ``/company/tools/zephyr-sdk-x.y.z``
-
-This allows the Zephyr build system to choose the correct version of the SDK,
-while allowing multiple Zephyr SDKs to be grouped together at a specific path.
+Toolchain variant variable
+   By default, the Zephyr build system assumes that the toolchain will be obtained
+   from the Zephyr SDK. You can enforce this by setting the environment variable
+   :envvar:`ZEPHYR_TOOLCHAIN_VARIANT` to ``zephyr``.
+
+SDK installation directory variable
+   If you install the Zephyr SDK outside any of the default locations (listed in
+   the operating system specific instructions below), you must register the Zephyr
```

Comment:
> you must register

is this a **must** ?
And since when ?

Reply:
@carlescufi , this is a question for you :)

Reply:
It´s a must if you do not use environment variables @tejlmand?

In order for the SDK to be found by the build system you need:
- To install it in a default location OR
- to register it with CMake OR
- to set the `ZEPHYR_SDK_INSTALL_DIR`

isn't this correct?

Reply:
As long as you don't state how `ZEPHYR_SDK_INSTALL_DIR` is set.
It is correct that it must be set if not installed in default search path or registered as a CMake package.

But it doesn't have to be set in environment. It may be set in the CMake cache using `-D`.
Another option is to use the [Zephyr Build Configuration CMake package](https://docs.zephyrproject.org/latest/build/zephyr_cmake_package.html#zephyr-build-configuration-cmake-package), which can be used on project level to ensure a given toolchain is used. For example this is what is done by NCS and the NCS Toolchain.
Using a build configuration CMake package allows a project to define a specific toolchain and remove the need for user interaction.

Note: I just discovered that @greg-fer only reformatted the text and actually didn't change the wording.

How about this:

> If you install the Zephyr SDK outside any of the default locations (listed in the operating system specific instructions below) and you want automatic discovery of the Zephyr SDK, then you must register the Zephyr SDK in the CMake package registry by running the setup script.
> If you decide not to register the Zephyr SDK in the CMake registry, then the :envvar:`ZEPHYR_SDK_INSTALL_DIR` can be used to point to the Zephyr SDK installation directory.

That way we don't state that you must use one of those two ways, but that you can use one of those.
And at this point I don't believe we need to mention all possible options, as that should be generally describe, see https://github.com/zephyrproject-rtos/zephyr/issues/40389

Reply:
Applied suggestion...